packages/core/component.ts

```diff
@@ -1,63 +1,60 @@
-export class Component extends HTMLElement {
-    static getName(): string {
-        throw new Error("Custom component should overload static getName method")
-    }
-
-    static get observedAttributes(): string[] {
-        return [];
-    }
-
-    isShadow: boolean = false;
-    root: HTMLDivElement;
-
-    createRoot() {
-        if (this.isShadow) {
-            this.attachShadow({mode: 'open'})
-        } else {
-            this.root = document.createElement('div');
-            this.root.style.display = 'contents';
-            this.appendChild(this.root);
-        }
-    }
-
-    beforeRender() {}
-
-    afterRender() {}
-
-    connected() {}
-
-    disconnected() {}
-
-    propChanged(name: string, oldValue: string, newValue: string) {}
-
-    render(root: ShadowRoot | HTMLDivElement) {}
-
-    callRender(): void {
-        this.beforeRender();
-        this.render(this.isShadow ? this.shadowRoot : this.root);
-        this.afterRender();
-    }
-
-    connectedCallback() {
-        this.createRoot();
-        this.callRender();
-
-        this.connected()
-    }
-
-    disconnectedCallback() {
-        this.disconnected()
-    }
-
-    attributeChangedCallback(name: string, oldValue: string, newValue: string) {
-        if (typeof oldValue != "undefined" && typeof newValue != "undefined") {
-            if (oldValue !== newValue) {
-                this.propChanged(name, oldValue, newValue)
-            }
-        }
-    }
-
-    update(): void {
-        this.callRender()
-    }
-}
\ No newline at end of file
+import {
+  BaseComponent
+} from './base-component';
+
+export class Component extends BaseComponent {
+  static getName(): string {
+    throw new Error("Custom component should overload static getName method")
+  }
+
+  /** @fixme is it never used? */
+  static get observedAttributes(): string[] {
+    return [];
+  }
+
+  isShadow: boolean = false;
+  root: HTMLDivElement;
+
+  createRoot() {
+    this.init()
+    if (this.isShadow) {
+      this.attachShadow({
+        mode: 'open'
+      })
+    } else {
+      this.root = document.createElement('div');
+      this.root.style.display = 'contents';
+      this.appendChild(this.root);
+    }
+  }
+
+  callRender(): void {
+    this.beforeRender();
+    this.render(this.isShadow ? this.shadowRoot : this.root);
+    this.afterRender();
+  }
+
+  connectedCallback() {
+    this.createRoot();
+    this.callRender();
+
+    this.connected()
+  }
+
+  disconnectedCallback() {
+    this.disconnected()
+  }
+
+  /** @fixme is it never used? */
```

Comment:
it's fired when html attribute for web-component is changed. this method is to perform shallow equals for old & new value and to call this.propChanged(our interface over web-components)

Reply:
I proposed to move everything to the base class.
And add some examples using this methods

Reply:
now you decomposed base class as class with overridable methods only with no implementation - this method contains imlementation

Reply:
ok, let's keep it as is and examples to the `micro-frontends-starter`

Reply:
we'll add it to the Component docs first of all; and to starter if you want